src/transformStateWithClones.js

```diff
@@ -7,7 +7,54 @@
  * @return {Object[]}
  */
 function transformStateWithClones(state, actions) {
-  // write code here
-}
+  const stagesOfModify = [];
+  const modifyingState = JSON.parse(JSON.stringify(state));
```

Comment:
no need to use deep copy in this case,you can just use spread operator to create a copy